### DISCUSSION TITLE
ArgoCD create job only change deployment

### DISCUSSION THREAD

**Discussion by milank78git** [argoproj/argo-cd#16194] [Q&A]
"I have a Job that I need to run only on the first initialization in ArgoCD or when there is a change in the deployment itself, not on refresh or hard refresh."

I tried but create  job if used hard-refresh and sync prune
ArgoCD v2.6.6+6d4de2e
apiVersion: batch/v1
kind: Job
metadata:
  labels:
  generateName: postdeployment-success-
  annotations:
    helm.sh/hook: post-install,post-upgrade
    argocd.argoproj.io/hook: PostSync
    argocd.argoproj.io/hook-delete-policy:  HookFailed
spec:
  template:
    metadata:
      annotations:

**Comment by crenshaw-dev** [comment]
Neither refresh nor hard refresh should trigger the hook as configured. 

I don't think hooks support specifying a single resource - if you prune an unrelated resource, that's considered a "sync," and the hook will run.

**Comment by milank78git** [comment]
So how to set the condition that the job starts only on new application deployment
Thank you very much for your help

**Reply by crenshaw-dev** [reply]
I probably wouldn't use a hook. An initContainer is probably a better fit.

**Comment by milank78git** [comment]
I need the job to run after a successful deployment and only during a new version deployment of the application, not during synchronization.
